server/channels/api4/channel.go

```diff
@@ -89,6 +90,16 @@ func createChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if channel.TeamId == "" {
+		c.SetInvalidParamWithDetails("team_id", i18n.T("api.channel.create_channel.missing_team_id.error"))
+		return
+	}
+
+	if channel.DisplayName == "" {
+		c.SetInvalidParamWithDetails("display_name", i18n.T("api.channel.create_channel.missing_display_name.error"))
+		return
+	}
```

Comment:
I'm not too keen on adding additional validation code on top of what `IsValid` already does. Can we use that method instead?

Reply:
Okay. On that note. For i18n. How do I get the translations for the other languages?

Reply:
Adding the validation to IsValid doesn't work in this situation. The code checks for permission to team beforehand. And it will fail there first.

Reply:
Good point on the permission check.